pkg/services/apiserver/builder/common.go

```diff
@@ -29,8 +29,7 @@ type APIGroupBuilder interface {
 		scheme *runtime.Scheme,
 		codecs serializer.CodecFactory,
 		optsGetter generic.RESTOptionsGetter,
-		desiredMode grafanarest.DualWriterMode,
-		reg prometheus.Registerer,
+		dualWrite grafanarest.DualWriteBuilder,
```

Comment:
two variables... not so bad, but this keeps growing and forces each implementation to know what they are all for